integration_test/go/testserver/testserver.go

```diff
@@ -174,10 +174,10 @@ func (s *testSrv) PingError(ctx context.Context, ping *testproto.PingRequest) (*
 
 func (s *testSrv) ContinueStream(ctx context.Context, req *testproto.ContinueStreamRequest) (*google_protobuf.Empty, error) {
 	s.streamsMutex.Lock()
-	defer s.streamsMutex.Unlock()
 	channel, ok := s.streams[req.GetStreamIdentifier()]
+	s.streamsMutex.Unlock()
 	if !ok {
-		return nil, grpc.Errorf(codes.NotFound, "stream identifier not found")
+		return nil, grpc.Errorf(codes.NotFound, fmt.Sprintf("stream identifier not found: %s", req.GetStreamIdentifier()))
```

Comment:
Nit: we should use the [`status` package](https://pkg.go.dev/google.golang.org/grpc/status#Errorf), this function has been deprecated for years.
```suggestion
		return nil, status.Errorf(codes.NotFound, "stream identifier not found: %s", req.GetStreamIdentifier())
```